cranelift/codegen/meta/src/shared/instructions.rs

```diff
@@ -1389,7 +1389,10 @@ pub(crate) fn define(
         )
         .operands_in(vec![c, x, y])
         .operands_out(vec![a])
-        .other_side_effects(true),
+        .other_side_effects(true)
+        // We can de-duplicate spectre selects since the side effect is
+        // idempotent.
+        .side_effects_okay_for_gvn(true),
```

Comment:
Do we actually need `select_spectre_guard` to have "side effects"? The important part is that it lowers to a conditional move whose output can't be speculated, right? Moving, duplicating, merging or removing it (as prevented by `other_side_effects(true)` shouldn't matter, right?

Reply:
I had this discussion with @cfallin (out of office right now) and the conclusion was that we probably didn't technically need it, but leaving it as-is was a nice way to be conservative here, since we want to be really careful with it.

I do plan to use `side_effects_okay_for_gvn` for some more stuff (eg `uadd_overflow_trap`) that is *not* movable (ie invalid to LICM hoist) but *is* GVN-able, so it isn't like this infrastructure is completely unnecessary.

Reply:
Maybe call it side_effects_idempotent and add a comment that it allows merging instructions into an identical instruction earlier in the runtime program order but doesn't allow moving?